enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
```

Comment:
- [ ] enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- [ ] enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- [ ]  enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [ ] enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ] enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern